src/collection/components/CollectionOrBundleEditMarcom.tsx

```diff
@@ -98,6 +105,25 @@ const CollectionOrBundleEditMarcom: FC<CollectionOrBundleEditMarcomProps & UserP
 		}
 	}, [collection.id, tHtml]);
 
+	const fetchKlascementPublishInfo = useCallback(async () => {
+		try {
+			const publishInfo = await KlascementService.getKlascementPublishInfoForCollection(
+				collection.id
+			);
+			setKlascementAltText(publishInfo?.alt_text);
+			setKlascementSourceText(publishInfo?.source_text);
+			setKlascementImageUrl(publishInfo?.image_url);
+			setKlascementId(publishInfo?.klascement_id ?? undefined);
+		} catch (err) {
+			console.error(
+				new CustomError('Failed to fetch klascement publish info from the database', err, {
+					collectionId: collection.id,
+				})
+			);
+			ToastService.danger(tHtml('Het ophalen van de klascement informatie is mislukt'));
+		}
+	}, [collection.id, tHtml]);
```

Comment:
Wrap this in a react-query wrapper so we get loading states, error states and caching
```
const {data: publishInfo, refetch: refetchPublishInfo, isLoading: isLoadingPublishInfo} = useGetKlascementPublishInfo(collection.id);
```

```
export const useGetKlascementPublishInfo = (collectionId: string): UseQueryResult<KlascementPublishInfo> => {
	return useQuery(
		[QUERY_KEYS.GET_KLASCEMENT_PUBLISH_INFO, collectionId],
		() => {
			return KlascementService.getKlascementPublishInfoForCollection(collectionId);
		}
	);
};
```